### ISSUE TITLE
compute sample metics before and after particles reduction

### ISSUE THREAD

**Issue by KseniaBastrakova** [ComputationalRadiationPhysics/particle_reduction#5]
It makes sense to calculate the sample metrics before and after the distribution. We need to consider weights for each item. There is good package in R for it.
[Weighted.Desc.Stat](https://cran.r-project.org/web/packages/Weighted.Desc.Stat/index.html)

**Comment by ax3l** [comment]
Just in case you need it, weighted binning in numpy's nd histograms as well as pandas + [statsmodels](https://www.statsmodels.org) might offer similar functionality in python [[SO](https://stackoverflow.com/questions/17689099/using-describe-with-weighted-data-mean-standard-deviation-median-quantil)].

But especially moments are not too hard to implement quickly oneself: [wiki](https://en.wikipedia.org/wiki/Moment_(mathematics))

**Comment by sbastrakov** [comment]
@ax3l you are right, moments should be easily available or in the worst case easy to implement. And that linked R package seem to offer nothing more (so by itself it should definitely not be a reason to switch to R). 

Additionally to that - and that is what is maybe more presented in R than python - we need metrics used in statistics, that fall into three categories:

- Distance between two *weighted* samples, at least 1D, better 3D and 6D as well (so far found only 1D Wasserstein in python iirc and used it for energies for the poster)
- Distance between two histograms
- Distance between two densities - here we can basically use any metric in functional space, but maybe some make more sense particularly for densities
